java/src/com/swiftnav/sbp/drivers/SBPDriverHTTP.java

```diff
@@ -0,0 +1,44 @@
+package com.swiftnav.sbp.drivers;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+
+public class SBPDriverHTTP extends SBPDriverStream {
+    private final URL url;
+    private final String uid;
+
+    public SBPDriverHTTP(String uid_, String url_) throws MalformedURLException {
+        url = new URL(url_);
+        uid = uid_;
+    }
+
+    @Override
+    protected void openStream() throws IOException {
+
+        System.out.println("Connecting to " + url.toString());
+
+        HttpURLConnection http = (HttpURLConnection)url.openConnection();
```

Comment:
You may want to set up some connection and read timeouts here as well.

http://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#setReadTimeout(int)

What happens if the connection fails in the app? Can you reinitiate the connection manually? I was looking to see if the JDK had a way of doing retries, but apparently not.